### PR TITLE
Added array and object support to is.equal()

### PR DESCRIPTION
**Pull request by moashour**
Hey, noticed that you wanted array and object support for `is.equal()`, so I built it out (adapting Unerscore's deep equality check). Hope you dig.

### CODE REVIEW

**Comment by moashour** [comment]
Sorry about the stupid .gitconfig. I was trying to get two github accounts to work on my machine.

**Comment by webdevbrian** [comment]
Is this going to be merged? Source looks great to me, and I have an immediate need for this on a few projects of mine. I was going to just pull this down and merge it myself, but I really think this should be added.

Thanks for the work @moashour ! Let's do this @jdalton 👍

**Comment by jdalton** [comment]
I'm not a fan of simply copying Underscore/Lodash which do it better anyways. The NIH bit of this lib is something I'm glossing over while I repo-sit so no this won't likely land on my watch because there's no reason not to use a modular lib like Lodash for these cases.

**Comment by webdevbrian** [comment]
Didn't you create lodash? In any event this is a pretty big need for many, so what's the proposed solution that jives with your way of thinking?

**Comment by jdalton** [comment]
> Didn't you create lodash?

Yep. You see the conflict.

> In any event this is a pretty big need for many

And there are lots of great solutions available.

**Comment by webdevbrian** [comment]
@jdalton Is it just that you don't have time to dig into this PR?

And of course there are a few solutions available (including just writing my own). I'd only assume others would agree, that this lib already doing equal checks should probably include this to limit redundancy. But via your comment in the lib ( TODO: Add object and array support ) I'm assuming you at least thought about this before.

I totally get that you have other things to do (we all do, lets be honest) - I guess what I'm asking at this point is, if you're "repo-sitting" as you put it, when can we expect this implementation to be in?

I see no real issue with @moashour's code, and opinons are great on it, but it'd be great to have a clear answer on what we can do, other than writing our own checks to fill in this gap of the lib.

Thanks

**Comment by ironmaniiith** [comment]
Well well, that TODO was added by @arasatasaygin

**Comment by webdevbrian** [comment]
@ironmaniiith Thanks for that ... I just saw dalton just edited his reply as well:

> I'm not a fan of simply copypasta Underscore/Lodash here.
> This won't likely land by me, while repo-sitting, because of my lib leanings.

Which makes sense -- but at least we know he won't be the one implementing it. I just wanted to comment on this and put this on his or @arasatasaygin 's radar that it's obviously needed, I just don't know the status of it (re: my comment with the TODO mention and @moashour 's work at the very least in the right direction)

File: is.js [hunk 1]
```diff
@@ -85,6 +85,75 @@
...
+        }
+        var keys = [],
+            key;
+        for(key in object) {
+            if(hasOwnProperty.call(object, key)) {
+                keys.push(key);
+            }
+        }
+        return keys;
+    };
```
Comment (jdalton) [inline is.js]: Could you factor out the `objectKeys` helper and avoid the keys length early exit.

File: .gitconfig [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[user]
+  name = Mohammad Ashour
+  email = mashour@sakani.com
```
Comment (jdalton) [inline .gitconfig]: Could you rebase this .gitconfig out.

**Comment by jdalton** [comment]
Closing as stale.

**Pull request closed without merging**